### PR TITLE
Refactor recent visualizations -> Postgres caching

### PR DESCRIPTION
**Pull request by JamesKunstle**
* adds tables with db_init for:
	- cntrb_per_file_query
	- pr_files_query
	- pr_response_query
	- repo_files_query
* refactors above queries to use cache_facade and write to postgres
* updates multiple visualizations to read from postgres_cache including:
	- both heatmaps
	- PR messaging response window

** I'm sorry the linter got run and committed, the major real changes are above.

### CODE REVIEW

**Comment by JamesKunstle** [comment]
@cdolfi From this branch, we'll merge to dev, hopefullly

File: 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py [hunk 1]
```diff
@@ -160,16 +160,22 @@ def repo_dropdown(repo_ids):
...
+        repolist=[repo_id],
+    )
+
+    logging.warning(f"DIRECTORY DROPDOWN - CACHE READ {df.shape[0]}")
 
     # test if there is data
     if df.empty:
         logging.warning(f"{VIZ_ID} DROPDOWN- NO DATA AVAILABLE")
-        return "NO DATA"
+        return dash.no_update, "NO DATA"
```
Comment (cdolfi) [inline 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py:178]: replace with 
`return ["Top Level Directory"], "Top Level Directory"`

File: 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py [hunk 2]
```diff
@@ -203,10 +213,54 @@ def directory_dropdown(repo_id):
 
     # add top level directory to the list of directories
     directories.insert(0, "Top Level Directory")
+    logging.warning(f"DIRECTORY DROPDOWN - FINISHED")
 
     return directories, "Top Level Directory"
 
 
+def multi_query_helper(repos):
```
Comment (cdolfi) [inline 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py:221]: for readability and consistency for helper functions in other visualizations move below create_figure

File: 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py [hunk 3]
```diff
@@ -177,11 +177,17 @@ def repo_dropdown(repo_ids):
...
+    logging.warning(f"DIRECTORY DROPDOWN - RETRIEVING FROM CACHE")
+    df = cf.retrieve_from_cache(
+        tablename=rfq.__name__,
+        repolist=[repo_id],
+    )
+
+    logging.warning(f"DIRECTORY DROPDOWN - CACHE READ {df.shape[0]}")
 
     # test if there is data
     if df.empty:
```
Comment (cdolfi) [inline 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py:193]: replace with
return ["Top Level Directory"], "Top Level Directory"

File: 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py [hunk 4]
```diff
@@ -177,11 +177,17 @@ def repo_dropdown(repo_ids):
...
+        logging.warning(f"DIRECTORY DROPDOWN - WAITING ON DATA TO BECOME AVAILABLE")
+        time.sleep(0.5)
+
+    logging.warning(f"DIRECTORY DROPDOWN - RETRIEVING FROM CACHE")
+    df = cf.retrieve_from_cache(
+        tablename=rfq.__name__,
+        repolist=[repo_id],
+    )
+
+    logging.warning(f"DIRECTORY DROPDOWN - CACHE READ {df.shape[0]}")
```
Comment (cdolfi) [inline 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py:190]: should shape still be here?

File: 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py [hunk 5]
```diff
@@ -160,16 +160,22 @@ def repo_dropdown(repo_ids):
...
+        logging.warning(f"DIRECTORY DROPDOWN - WAITING ON DATA TO BECOME AVAILABLE")
+        time.sleep(0.5)
+
+    logging.warning(f"DIRECTORY DROPDOWN - RETRIEVING FROM CACHE")
+    df = cf.retrieve_from_cache(
+        tablename=rfq.__name__,
+        repolist=[repo_id],
+    )
+
+    logging.warning(f"DIRECTORY DROPDOWN - CACHE READ {df.shape[0]}")
```
Comment (cdolfi) [inline 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py:173]: should shape still be here?

File: 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py [hunk 6]
```diff
@@ -136,6 +136,7 @@
     ],
 )
 
+
```
Comment (cdolfi) [inline 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py:139]: remove extra blank line

File: 8Knot/pages/contributors/visualizations/active_drifting_contributors.py [hunk 7]
```diff
@@ -292,9 +286,7 @@ def create_figure(df_status: pd.DataFrame, interval):
         )
 
         # edit hover values
-        fig.update_traces(
-            hovertemplate=hover + "<br>Contributors: %{y}<br>" + "<extra></extra>"
-        )
+        fig.update_traces(hovertemplate=hover + "<br>Contributors: %{y}<br>" + "<extra></extra>")
```
Comment (cdolfi) [inline 8Knot/pages/contributors/visualizations/active_drifting_contributors.py:289]: ugh i hate this change but i know its the linter

File: 8Knot/queries/repo_files_query.py [hunk 8]
```diff
@@ -48,7 +47,7 @@ def repo_files_query(self, repos):
                     repo r
                 WHERE
                     rl.repo_id = r.repo_id AND
```
Comment (cdolfi) [inline 8Knot/queries/repo_files_query.py:49]: still showing old query

**Review by cdolfi — CHANGES_REQUESTED** [review]
mostly small things and then will be good to go

File: 8Knot/cache_manager/cache_facade.py [hunk 9]
```diff
@@ -152,16 +152,16 @@ def get_uncached(func_name: str, repolist: list[int]) -> list[int]:  # or None
             return not_cached
 
 
-def caching_wrapper(func_name: str, query: str, repolist: list[int]) -> None:
+def caching_wrapper(func_name: str, query: str, repolist: list[int], n_repolist_uses=1) -> None:
```
Comment (cdolfi) [inline 8Knot/cache_manager/cache_facade.py:155]: whats this change?

File: 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py (hunk 1 above)
Comment (JamesKunstle) [reply 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py:178]: ++ updated

File: 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py (hunk 2 above)
Comment (JamesKunstle) [reply 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py:221]: ++ updated

File: 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py (hunk 3 above)
Comment (JamesKunstle) [reply 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py:193]: ++ updated

File: 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py (hunk 4 above)
Comment (JamesKunstle) [reply 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py:190]: ++ updated

File: 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py (hunk 5 above)
Comment (JamesKunstle) [reply 8Knot/pages/codebase/visualizations/cntrb_file_heatmap.py:173]: ++ updated

File: 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py (hunk 6 above)
Comment (JamesKunstle) [reply 8Knot/pages/codebase/visualizations/contribution_file_heatmap.py:139]: that's inserted between functions by the linter- it puts two lines in between functions

File: 8Knot/pages/contributors/visualizations/active_drifting_contributors.py (hunk 7 above)
Comment (JamesKunstle) [reply 8Knot/pages/contributors/visualizations/active_drifting_contributors.py:289]: yeah I agree

File: 8Knot/queries/repo_files_query.py (hunk 8 above)
Comment (JamesKunstle) [reply 8Knot/queries/repo_files_query.py:49]: ++ updated

File: 8Knot/cache_manager/cache_facade.py (hunk 9 above)
Comment (JamesKunstle) [reply 8Knot/cache_manager/cache_facade.py:155]: In the sql query for repo_files, the repolist is used twice, so we have to pass it as a 'var' to psycopg2 twice. I'm going to generalize this in the near future, but this works for the queries we currently have.

Comment (cdolfi) [reply 8Knot/cache_manager/cache_facade.py:155]: Makes sense, just didnt want to approve something that I didnt understand. Add comment in to explain temp step

Comment (JamesKunstle) [reply 8Knot/cache_manager/cache_facade.py:155]: It's already below in the Args list

Comment (cdolfi) [reply 8Knot/cache_manager/cache_facade.py:155]: ah i see now, I misread the initial explanation at first

**Review by cdolfi — COMMENTED** [review]
lgtm at this step, going to run locally to see if I can get anywhere w that bug

**Comment by cdolfi** [comment]
TODOs before this merge (or open another issue) or to dev: 
- go through types on db_init 
- test changes to queries to make sure they match the results of the old one

**Comment by JamesKunstle** [comment]
#584 takes changes suggested in this PR and switches the contributing branch. This PR can be closed because changes requested live in #584

**Pull request closed without merging**